torch/csrc/autograd/engine.cpp

```diff
@@ -393,35 +397,29 @@ void Engine::reentrant_thread_init() {
 }
 
 void Engine::thread_on_exception(
-    std::shared_ptr<GraphTask>& graph_task,
```

Comment:
Why do you remove the reference here?

Reply:
The reason is that with the const ref, the GraphTask can be deallocated underneath and thats why we had a comment in `set_exception` about not touching `*this`. This just ensures we pass a copy in so `thread_on_exception` keeps the pointer alive. I felt its just safe doing this instead of relying on developers to avoid touching `*this` after setError is called.